model-optimizer/extensions/front/tf/FifoQueueDequeueCut.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from collections import defaultdict
+
+from extensions.front.output_cut import OutputCut
+from extensions.front.user_data_repack import UserDataRepack
+from mo.front.common.replacement import FrontReplacementPattern
+from mo.front.extractor import add_input_ops
+from mo.graph.graph import Graph
+from mo.middle.passes.convert_data_type import np_data_type_to_precision, SUPPORTED_DATA_TYPES
+from mo.utils.error import Error
+
+
+class FifoQueueDequeueCut(FrontReplacementPattern):
+    """
+    Cuts FifoQueue -> QueueDequeue pattern
+    in order to enable Out Of the Box (OOB) usage.
+    Pass is run only if user didn't specify any inputs names and shapes.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['cmd_params'].input is None]
+
+    def run_before(self):
+        return [OutputCut]
+
+    def run_after(self):
+        return [UserDataRepack]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        fifo_qd_shapes = defaultdict(list)
+        for node in graph.get_op_nodes():
+            if node.op not in ["QueueDequeue", "QueueDequeueV2"]:
+                continue
+
+            fifo_qd_name = node.soft_get('name', node.id)
+            for port in node.out_nodes().keys():
```

Comment:
Can we use node.out_ports() here?
And port.idx if index is needed

Reply:
`node.out_ports()` returns an empty dict, because `out_ports_count` attribute is not defined in `QueueDequeue` extractor. `QueueDequeue` `_output_shapes` attrubute does not match a number of real consumers. Using `out_nodes()` is much easier to handle this case and to set a correct shape for `Parameter` node

Reply:
Okay, got it.